### PR TITLE
Migrate to API local store cache

### LINKED ISSUE #1486: Migrate any usage of localstorage for user interaction to api user storage
We are using flag now for
- Sharing explainer
- Survey dismissed

We should migrate them to user storage object, exposed on the UserContext. Linked to https://github.com/ChainSafe/files-api/pull/1427

### PR DESCRIPTION
**Pull request by RyRy79261**
closes #1486

### CODE REVIEW

**Review by FSM1 — DISMISSED** [review]
I am unable to get the Survey banner and Sharing modal to appear. Tested with a new account, in a private browser session, and no luck.

**Comment by RyRy79261** [comment]
@FSM1 The survey only shows for users that are older than a week

**Comment by RyRy79261** [comment]
And the sharing dialog displayed the intro modal for me: 
image

**Comment by FSM1** [comment]
> @FSM1 The survey only shows for users that are older than a week

ahhh gotcha, that makes sense... forgot about that rule 🙈

File: packages/files-ui/src/Contexts/UserContext.tsx [hunk 1]
```diff
@@ -59,6 +81,30 @@ const UserProvider = ({ children }: UserContextProps) => {
...
+      if (!localStore) {
+        // Fetch
+        try {
+          const fetched = await filesApiClient.getUserLocalStore()
+          if (!fetched) {
+            _setLocalStore({})
+          } else {
+            _setLocalStore(fetched)
+          }
+        } catch(error) {
```
Comment (tanmoyAtb) [inline packages/files-ui/src/Contexts/UserContext.tsx:95]:
```suggestion
        } catch(error) {
          console.error(error)
```

**Review by tanmoyAtb — APPROVED** [review]
Good to go !

**Review by FSM1 — APPROVED** [review]

**Pull request merged**